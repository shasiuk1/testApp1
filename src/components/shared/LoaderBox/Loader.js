/** @format */

import React from 'react';
import PropTypes from 'prop-types';
import cssModules from 'react-css-modules';
import types from 'SRC/types';
import Spinner from 'SRC/components/shared/Spinner';

import styles from './index.scss';

const Loader = ({ children, style }) => (
  <div styleName="loader" style={style}>
    <div styleName="loader-icon">
      <Spinner />
    </div>

    <div>{children}</div>
  </div>
);

Loader.propTypes = {
  children: types.children,
  style: PropTypes.object,
};

Loader.defaultProps = {
  children: 'Please wait, loading..',
  style: {},
};

export default cssModules(Loader, styles);
