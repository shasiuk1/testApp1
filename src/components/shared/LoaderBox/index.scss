@import '~SRC/styles/variables';
@import '~SRC/styles/mixins';

.loader {
  display: flex;
  align-items: center;
  padding: 1.5em;
  margin: 0 auto;
  font-size: $text-size;
  background-color: $color-gray-lighter;
}

.loader-icon {
  margin-right: 1rem;
}
