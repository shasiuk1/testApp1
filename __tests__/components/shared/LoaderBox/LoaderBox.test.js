/** @format */

import { shallow } from 'enzyme/build';
import React from 'react';
import LoaderBox from 'SRC/components/shared/LoaderBox';
import toJson from 'enzyme-to-json';

describe('test LoaderBox', () => {
  it('component is defined', () => {
    const wrapper = shallow(<LoaderBox />);
    expect(wrapper.find(LoaderBox)).toBeDefined();
  });

  it('component matches snapshot', () => {
    const wrapper = shallow(<LoaderBox />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('component has 1 processing icon', () => {
    const wrapper = shallow(<LoaderBox />);
    const icon = wrapper.find('.loader-icon');
    expect(icon.length).toBe(1);
  });
});
