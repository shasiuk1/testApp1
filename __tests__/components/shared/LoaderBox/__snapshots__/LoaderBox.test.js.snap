// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LoaderBox component matches snapshot 1`] = `
<div
  className="loader"
  style={Object {}}
>
  <div
    className="loader-icon"
  >
    <Spinner />
  </div>
  <div>
    Please wait, loading..
  </div>
</div>
`;
